src/dict.c

```diff
@@ -510,20 +510,23 @@ void dictRelease(dict *d)
 dictEntry *dictFind(dict *d, const void *key)
 {
     dictEntry *he;
-    uint64_t h, idx, table;
+    uint64_t h;
+    long idx_0, idx_1;
 
     if (dictSize(d) == 0) return NULL; /* dict is empty */
     if (dictIsRehashing(d)) _dictRehashStep(d);
     h = dictHashKey(d, key);
-    for (table = 0; table <= 1; table++) {
-        idx = h & DICTHT_SIZE_MASK(d->ht_size_exp[table]);
-        he = d->ht_table[table][idx];
```

Comment:
I think you could get the same result with a smaller change right here. If you just add:
```
if (dictIsRehashing(d) && d->rehashidx >= idx) continue;
```